ocaml-lsp-server/src/document_store.mli

```diff
@@ -2,16 +2,22 @@ open Import
 
 type t
 
-val make : unit -> t
+val make : _ Server.t Fdecl.t -> Fiber.Pool.t -> t
 
-val put : t -> Document.t -> unit
+val open_ : t -> Document.t -> unit Fiber.t
+
+val update : t -> Uri.t -> f:(Document.t -> Document.t) -> Document.t
 
 val get : t -> Uri.t -> Document.t
 
 val get_opt : t -> Uri.t -> Document.t option
 
 val remove_document : t -> Uri.t -> unit Fiber.t
 
+val unregister_promotions : t -> Uri.t list -> unit Fiber.t
+
+val register_promotions : t -> Uri.t list -> unit Fiber.t
```

Comment:
it's not obvious why `register_promotions` would take a param of type `Uri.t list`; a named arg such as `docs_to_promote` would help, in my opinion. 

I also saw other places where `Uri.t` is passed rather than a document. I think it could help reading code if the argument is a named one, say, `doc`.

Reply:
> it's not obvious why register_promotions would take a param of type Uri.t list; a named arg such as docs_to_promote would help, in my opinion.

Can't say i'm a fan of `doc_to_promote`.

> I also saw other places where Uri.t is passed to represent a document. I think it could help reading code if the argument is a named one, say, doc.

I suppose this is acceptable. `doc` and `docs` would be consistent. Though I don't see the big problem since `Uri.t` is already fairly descriptive unlike something like `string` or `int`. Let's save it for another PR that refactors the interface this module more thoroughly.